### DISCUSSION TITLE
Splash Screen while server is generating?

### DISCUSSION THREAD

**Discussion by yuriburk** [vercel/next.js#15963] [Help] [answered]
How do I show a Splash Screen (any component actually) while server is generating the HTML to pages/Index (getStaticProps)?

I know that can be done with Router when navigate to another page, but It doesn't work on the home page (I want to show it on site loading). Any help it's appreciated. Thanks.

**Comment by qnxdev** [comment]
{!props.data && <SplashScreen/>}

**Comment by yuriburk** [comment, marked as answer]
Answered https://github.com/vercel/next.js/discussions/15974
